baseplate/server/healthcheck.py

```diff
@@ -60,15 +67,30 @@ def parse_args() -> argparse.Namespace:
         default=Endpoint("localhost:9090"),
         help="The endpoint to find the service on.",
     )
+    parser.add_argument(
```

Comment:
The argument parsing could be made a little more user-friendly with a few tweaks:

## use choices= to give the parser better docs

Add `choices=[probe.lower() for probe in IsHealthyProbe._NAMES_TO_VALUES]` so that the help formatter can print out the available options to users and also forces validation into the argparse code instead of ours. 

## rely on argparse doing validation

With that, you can drop `_convert_probes` and make the for loop on L92 look like:

```python
for probe in args.probes:
    checker(args.endpoint, IsHealthyProbe._NAMES_TO_VALUES[probe.upper()])
```

## make it safer to do the arguments in any order

I think making it one multi-argument flag like this may make parsing kinda hard. If you do `--probes` before the two positional arguments (which on its face seems reasonable) it makes the parser not understand the positional arguments:

```
ubuntu@reddit:~/src/baseplate.py (baseplate.py)$ baseplate-healthcheck --probes readiness thrift localhost:9002
usage: baseplate-healthcheck [-h]
                             [--probes {readiness,liveness,startup} [{readiness,liveness,startup} ...]]
                             {thrift,wsgi} endpoint
baseplate-healthcheck: error: argument --probes: invalid choice: 'thrift' (choose from 'readiness', 'liveness', 'startup')
ubuntu@reddit:~/src/baseplate.py (baseplate.py)$ baseplate-healthcheck thrift localhost:9002 --probes readiness
OK!
```

maybe instead it can be `nargs="?"` and use `action="append"` to append values to the list? then you'd use multiple `--prompt=onething` flags to do multiple probes but the parsing would be unambiguous. Not sure how to get rid of the default in that case though.

Reply:
`action="append"` cannot handle default properly. when using that the default is always in the stored slice.

I think keeping default work is more important than supporting multiple probes, so I'll change it to only support a single probe instead.

Reply:
Done.